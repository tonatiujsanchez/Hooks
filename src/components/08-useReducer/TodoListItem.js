

import PropTypes from 'prop-types'


const TodoListItem = ({ todo, idx, handleToggle, handleDelete }) => {
    return (
        <li className="list-group-item d-flex justify-content-between align-items-center">
            <p
                className={`text-center m-0 todo-p ${todo.done ? 'complete' : ''}`}
                onClick={() => handleToggle(todo.id)}>
                {idx + 1}. {todo.desc}
            </p>
            <div>
                <button
                    onClick={() => handleDelete(todo.id)}
                    className='btn btn-danger'>
                    Borrar
                </button>
            </div>
        </li>
    )
}

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    idx: PropTypes.number.isRequired,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired
}

export default TodoListItem