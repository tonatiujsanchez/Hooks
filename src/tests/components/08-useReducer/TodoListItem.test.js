import { shallow } from "enzyme"
import TodoListItem from "../../../components/08-useReducer/TodoListItem"
import { demoTodos } from "../../fixtures/demoTodos"





describe('Pruebas en <TodoListItem />', () => { 

    const index = 1
    const todo = demoTodos[index]
    const handleDelete = jest.fn()
    const handleToggle = jest.fn()

    const wrapper = shallow(
        <TodoListItem 
            todo={ todo } 
            idx={ index }
            handleToggle={ handleToggle }
            handleDelete={ handleDelete } />
    )
   

    test('Debe de mostrarse correctamente', () => { 

        expect( wrapper ).toMatchSnapshot()

    })


     test('Debe de llamar la funcion handleDelete', () => {

        const button = wrapper.find('button')
        button.simulate('click', todo.id)


        expect( handleDelete ).toHaveBeenCalled()
        expect( handleDelete ).toHaveBeenCalledWith( todo.id )


    })


    test('Debe de llamar la funcion handleToggle', () => { 
        const parrafo = wrapper.find('p')
        parrafo.simulate('click', todo.id)

        expect( handleToggle ).toHaveBeenCalled()
        expect( handleToggle ).toHaveBeenCalledWith( todo.id )

    })


    test('Debe de mostrar el texto correctamente', () => { 

        const parrafo = wrapper.find('p')
        parrafo.simulate('click', todo.id)

        expect( parrafo.text().includes(todo.desc) ).toBe(true)
        expect( parrafo.text().includes( index + 1 ) ).toBe( true )

        expect( parrafo.text() ).toBe(`${ index + 1 }. ${ todo.desc }`)

    })


    test('Debe de tener la clase complete si todo.done = true', () => { 

        todo.done = true

        const wrapper = shallow(
            <TodoListItem 
                todo={ todo } 
                idx={index}
                handleToggle={ handleToggle }
                handleDelete={ handleDelete } />
        )

        const parrafo = wrapper.find('p')
        expect( parrafo.hasClass('complete') ).toBe(true)

    })
 })