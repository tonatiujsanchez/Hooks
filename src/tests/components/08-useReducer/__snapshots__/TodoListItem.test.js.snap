// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> Debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-center"
>
  <p
    className="text-center m-0 todo-p "
    onClick={[Function]}
  >
    2
    . 
    Aprender Next
  </p>
  <div>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Borrar
    </button>
  </div>
</li>
`;
